### PR TITLE
Fix requirement being used before installing

### LINKED ISSUE #51: Unable to install module ufpy
Unable to install module ufpy in PyCharm CE on MacOS with last version of pip and ARM64.

```          
from requests import get
      ModuleNotFoundError: No module named 'requests'
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip
```

- #52

### PR DESCRIPTION
**Pull request by n0n1m**
Fixes #51

## Summary by Sourcery

Fix the requirement of using the package version before installation by introducing a function to derive the version dynamically from the source code.

Bug Fixes:
- Fix the issue of using the package version before it is installed by deriving the version dynamically from the source code.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -15,6 +14,33 @@
...
+
+    if not version:
+        raise RuntimeError('version is not set')
+
+    if version.endswith(('a', 'b', 'rc')):
+        # append version identifier based on commit count
+        try:
+            import subprocess
+
+            p = subprocess.Popen(['git', 'rev-list', '--count', 'HEAD'], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```
[the remark this answers is not in the document]
Comment (n0n1m) [reply setup.py:30]: Yea, on second thought this part of function can be removed

File: setup.py [hunk 2]
```diff
@@ -15,6 +14,33 @@
 project_name = 'ufpy'
 github_url = f'https://github.com/{organization_name}/{project_name}'
 
+def derive_version() -> str: # this function is stolen from discord.py
```
[the remark this answers is not in the document]
Comment (n0n1m) [reply setup.py:17]: > 1. If dynamic versioning isn't critical, revert to the original method of importing the version directly:
> 
> ```python
> from ufpy import __version__
> ```

This solution not working is the reason this pull request exists

**Comment by bleudev** [comment]
@n0n1m Thank you for your work!

**Comment by Alexandr153** [comment]
pls try my method. I'm not saying that it will definitely help, but there have been a lot of changes with requests in recent versions

Снимок экрана 2024-08-31 в 22 32 45

**Comment by n0n1m** [comment]
> pls try my method. I'm not saying that it will definitely help, but there have been a lot of changes with requests in recent versions
> 
> Снимок экрана 2024-08-31 в 22 32 45

it is using the most recent version of requests

**Comment by n0n1m** [comment]
the problem is it is used before being installed

**Comment by n0n1m** [comment]
just try `pip install git+https://github.com/honey-team/ufpy.git`

**Comment by n0n1m** [comment]
it should fix an issue

**Comment by n0n1m** [comment]
also @bleudev, did you publish this commit to pypi?

**Comment by bleudev** [comment]
> also @bleudev, did you publish this commit to pypi?

No, i will publish new version when i will done new README.

**Comment by bleudev** [comment]
> just try `pip install git+https://github.com/honey-team/ufpy.git`

@Alexandr153 Can you try this? If the problem will be solved, we can close your PR.

**Comment by n0n1m** [comment]
> > just try `pip install git+https://github.com/honey-team/ufpy.git`
> 
> @Alexandr153 Can you try this? If the problem will be solved, we can close your PR.

You already closed it. Also you closed issue by merging this PR

**Comment by n0n1m** [comment]
I have tested PyPI version and this version in brand new venv. PyPI one threw this error and this installed fine

**Comment by n0n1m** [comment]
In fact, PyPI version doesn't install even with requests installed

**Comment by n0n1m** [comment]
That's why I think you should change version number to 0.2.1 (or smth) and publish it

**Pull request merged**